src/main/java/hudson/plugins/gradle/BuildScanAction.java

```diff
@@ -2,6 +2,12 @@
 
 import hudson.model.Action;
 
+/**
+ * @deprecated This class remains only for backwards compatibility so that actions persisted for old builds will
+ * continue to render correctly. At some point this should probably be removed.
+ *
+ * @see BuildScansAction
```

Comment:
Can you do the data migration of this class to the new class? Some hints here: https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility

Reply:
I did some refactoring here, but now I'm not sure there's a simple solution. The problem here is that the arity has changed. The old implementation supported multiple actions per build, but the new does not. I'm not sure there's a simple way to take n `BuildScanAction` and turn them into a single `BuildScansAction`. As it is now, my naive refactor will create multiple actions again for old builds which might result in a poor UI experience.

Reply:
There is the possibility to use an `Initializer`: https://github.com/jenkinsci/run-selector-plugin/blob/854deb121eace1d0f4a90664a6d2d2ad733e0d61/src/main/java/hudson/plugins/copyartifact/CopyArtifact.java#L148-L172

Please also add some test for the migration in `CompatibilityTest`.

Reply:
I'm concerned how costly it might be to go back and migrate this for all build. We aren't just migrating project config, these are actions attached to _actual_ build invocations. Going back and updating the entire Jenkins history might not be worth the effort here.

Reply:
OK, I am fine with leaving it the way it is now - i.e. the name of the action does not change, we only store a list instead of a single element. We could change the name as well (via `Items.XSTREAM2.addCompatibilityAlias`), though I don't think it is necessary.

We still need a test. I think currently it won't work since the list will be null when loaded from an old version of `BuildScanAction`.

Reply:
Yeah, as I mentioned, I had initially done that and implemented `readResolve()` to populate the list property if `scanUrl` was non-null but that breaks down because of the arity issue. We would create multiple of these actions which would then render oddly, as the updated UI assumes 0..n build scans.

I've simply reverted back to the initial compatibility method which just keeps the old implementation around. So for old builds, we'll use the old action and UI, and for new builds we'll use the new one. I simply don't have a better idea here that doesn't include a costly migration.

Reply:
Implementation is per our discussion. I've also added test coverage for compatibility.